### ISSUE TITLE
Bounded polymorphic function causes error if imported

### ISSUE THREAD

**Issue by Macil** [facebook/flow#1140]
fooer.js

``` js
/* @flow */

type Fooish = {bar(): void};

export default function fooer<T: Fooish>(fooFn: () => T) {
  const f = fooFn();
  f.bar();
}
```

test.js

``` js
/* @flow */

import fooer from './fooer';
class Foo {
  bar() {}
}
fooer(() => new Foo());
```

This causes the following Flow output:

```
[chris /tmp/foo]$ flow version
Flow, a static type checker for JavaScript, version 0.19.0
[chris /tmp/foo]$ flow
test.js:7
  7: fooer(() => new Foo());
     ^^^^^^^^^^^^^^^^^^^^^^ function call
  7:   f.bar();
       ^ object type. This type is incompatible with. See: fooer.js:7
global object


Found 1 error
```

However, if `fooer` is defined in the same file instead of being imported, then there are no Flow errors:

``` js
/* @flow */

type Fooish = {bar(): void};
function fooer<T: Fooish>(fooFn: () => T) {
  const f = fooFn();
  f.bar();
}
class Foo {
  bar() {}
}
fooer(() => new Foo());
```

**Comment by AgentME** [comment]
Another thing that makes the error disappear is changing the Fooish type:

```js
type Fooish = {bar: Function};
```

**Comment by billschaller** [comment]
@mroch This appears to be resolved - checked 0.41
